### PR TITLE
[release/3.1] Quarantine CanChangeEmailOnlyIfEmailSame test

### PR DESCRIPTION
**Pull request by halter73**
Investigation issue: #25716

### CODE REVIEW

File: src/Identity/Specification.Tests/src/Microsoft.AspNetCore.Identity.Specification.Tests.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore.Identity" />
+    <Reference Include="Microsoft.AspNetCore.Testing" />
```
Comment (halter73) [inline src/Identity/Specification.Tests/src/Microsoft.AspNetCore.Identity.Specification.Tests.csproj:15]: Below, there's `<SuppressBaselineReference Include="Microsoft.AspNetCore.Testing" />`. I'm not sure if that needs to be removed.

Comment (JunTaoLuo) [reply src/Identity/Specification.Tests/src/Microsoft.AspNetCore.Identity.Specification.Tests.csproj:15]: Yea remove it. This should just be a revert of https://github.com/dotnet/aspnetcore/commit/d10a352e8caec866c4107b2fd4fdc5ce49f7a740

**Review by JunTaoLuo — APPROVED** [review]
Once the suppressbaselinereference is removed

**Review by HaoK — APPROVED** [review]

**Comment by halter73** [comment]
@JunTaoLuo Are you sure it's OK to add a Microsoft.AspNetCore.Testing dependency to this project?

Looking at https://github.com/dotnet/aspnetcore/pull/18790, it seems the problem is that this adds an unshipped dependency to https://www.nuget.org/packages/Microsoft.AspNetCore.Identity.Specification.Tests/. I didn't realize that we shipped Microsoft.AspNetCore.Identity.Specification.Tests as a NuGet package when I first opened this PR. It makes sense, since we do the same thing for DI. It just didn't occur to me.

**Comment by JunTaoLuo** [comment]
Actually you are right, oops brain fart, yea we totally don't want to do this in 3.1. The dependency was removed for a reason.

**Comment by halter73** [comment]
Should we just delete the test until someone investigates it since we cannot quarantine it? Looking at https://github.com/dotnet/aspnetcore-internal/issues/1766, it seems CanChangeEmailOnlyIfEmailSame has been pretty flaky for well over a year. Shipping a flaky test doesn't seem like a good idea.

**Comment by JunTaoLuo** [comment]
Sure, file an issue to track the test investigation.

**Comment by HaoK** [comment]
Yeah just delete the test, there's no harm in that

**Comment by halter73** [comment]
I already created an issue at https://github.com/dotnet/aspnetcore/issues/25716. There's a very similar issue for a different test in the same class at https://github.com/dotnet/aspnetcore-internal/issues/2674. Both threw something similar to:

```
System.Security.Cryptography.CryptographicException : An error occurred while trying to encrypt the provided data. Refer to the inner exception for more information.
---- System.IO.IOException : The process cannot access the file 'C:\Users\vsagent\AppData\Local\ASP.NET\DataProtection-Keys\key-3b11648b-e3fc-4f12-a9db-1607a45973df.xml' because it is being used by another process.
```

It seems like there might be a real issue with using DataProtection in parallel, and since we probably don't want to delete everything in UserManagerSpecificationTests, I'm going to leave it.

**Pull request closed without merging**